### DISCUSSION TITLE
Any example of how to select and copy using the API?

### DISCUSSION THREAD

**Discussion by devmark404** [Orama-Interactive/Pixelorama#1101] [Q&A] [answered]
I would like to select and copy the first 9 pixels that are within the white border of the current layer

I am using the following code, but the first 9 pixels are not selected and copied

https://www.oramainteractive.com/Pixelorama-Docs/extension_system/extension_api/#method-descriptions-7

```
extension_api.selection.select_rect(Rect2(0, 0, 1, 9), 0)
extension_api.selection.copy()
```

![copy pixelorama](https://github.com/user-attachments/assets/bc5d0bea-ac4d-429b-acae-991cf7081682)

Español:
Quisiera seleccionar y copiar los primeros 9 píxeles que están dentro del borde blanco de la capa actual

Estoy usando el siguiente código, pero no se seleccionan y no se copian los primeros 

```
extension_api.selection.select_rect(Rect2(0, 0, 1, 9), 0)
extension_api.selection.copy()
```

**Comment by Variable-ind** [comment, marked as answer]
Hy, what happens when you use the paste shortcut (Ctrl + V).
I tried this but it seems to work for me.


https://github.com/user-attachments/assets/f4d9f44d-6885-4fea-93cc-fb652c6fb313

```swift
extends Node

var extension_api: Node
var selection_node: Node  ## Used to interact with the selection directly, instead of through api

# This script acts as a setup for the extension
func _enter_tree() -> void:
	## NOTE: use get_node_or_null("/root/ExtensionsApi") to access api.
	extension_api = get_node_or_null("/root/ExtensionsApi")
	extension_api.menu.add_menu_item(extension_api.menu.SELECT, "Copy 9 Pixels", self)
	## NOTE: You may want to get some other properties of the selection e.g size.
	## for the moment those aren't exposed to API,
	## so you have to interact with selection directly.
	selection_node = extension_api.general.get_global().canvas.selection
	## NOTE Now you can use functions given in pixelorama's res://src/UI/Canvas/Selection.gd file.


func menu_item_clicked():
	extension_api.selection.select_rect(Rect2(0, 0, 1, 9), 0)  ## Copies 9 pixels (From Top to Bottom)
	## NOTE: (Other helpful examples)
	#extension_api.selection.select_rect(Rect2(0, 0, 1, 9), 0)  ## Copies 9 pixels (From Left to Right)
	#extension_api.selection.select_rect(Rect2(0, 0, 3, 3), 0)  ## Copies 9 pixels (In a Square area)

	## NOTE: There is another option called cut (which is just copy + delete original)
	## in the api, the "cut" feature is split into two parts (copy + delete_content).
	extension_api.selection.copy()
	#extension_api.selection.delete_content()  ## Uncomment this line to cut instead of copy.

	## PRINT the selection size
	var selection_size: Rect2i = selection_node.big_bounding_rectangle
	extension_api.dialog.show_error(str(selection_size))


func _input(event: InputEvent) -> void:
	if event is InputEventKey:
		if event.physical_keycode == KEY_A:
			selection_node.transform_content_confirm()

```

**Reply by devmark404** [reply]
It worked now, it seems you have to restart pixelorama when you make changes to the plugin,

It doesn't matter if you deactivate and activate the plugin again you have to restart pixelorama or it won't recognize the changes

**Reply by Variable-ind** [reply]
Yeah, that's an issue from 1.0 onwards, not sure what causes it though.

**Reply by devmark404** [reply]
> Yeah, that's an issue from 1.0 onwards, not sure what causes it though.

I see

Another question, could inputs be added to the API to be able to change a plugin variable?

**Reply by Variable-ind** [reply]
Not sure what mean but I think it should be possible. Can you give me an example of what you mean by "plugin variable"?

**Reply by devmark404** [reply]
> Not sure what mean but I think it should be possible. Can you give me an example of what you mean by "plugin variable"?

For example,

If in my code I have a variable for the size of the selection, I want to be able to change that variable from pixelorama using a text input or a slider,

And finally press a button to confirm the size

**Reply by Variable-ind** [reply]
> For example,
> 
> If in my code I have a variable for the size of the selection, I want to be able to change that variable from pixelorama using a text input or a slider,
> 
> And finally press a button to confirm the size

It can't be done through Api at the moment, but it can still be done.
i've updated my code above.
(I will probably add more parameters and functions to the Api too for the future version 1.0.3)

Also, if you want to change the size, there is a resize() function available in the api
